keras_hub/src/models/mobilenetv5/mobilenetv5_blocks.py

```diff
@@ -0,0 +1,894 @@
+                channel_axis=self.channel_axis,
+                dtype=self.dtype_policy,
+            )
+            if se_layer
+            else (lambda x, training=False: x)
+        )
+        self.conv_pwl = ConvNormAct(
+            out_chs,
+            pw_kernel_size,
+            pad_type=pad_type_internal,
+            apply_act=False,
+            norm_layer=norm_layer,
+            act_layer=act_layer,
+            bias=use_bias,
+            data_format=self.data_format,
+            channel_axis=self.channel_axis,
+            dtype=self.dtype_policy,
+        )
+        self.drop_path = (
+            DropPath(drop_path_rate, dtype=self.dtype_policy)
+            if drop_path_rate > 0.0
+            else (lambda x, training=False: x)
+        )
+
+    def build(self, input_shape):
+        self.conv_exp.build(input_shape)
+        conv_exp_output_shape = self.conv_exp.compute_output_shape(input_shape)
+        if hasattr(self.se, "build"):
+            self.se.build(conv_exp_output_shape)
+        self.conv_pwl.build(conv_exp_output_shape)
+        self.built = True
+
+    def call(self, x, training=False):
+        shortcut = x
+        x = self.conv_exp(x, training=training)
+        x = self.se(x, training=training)
+        x = self.conv_pwl(x, training=training)
+        if self.has_skip:
+            x = self.drop_path(x, training=training) + shortcut
+        return x
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "in_chs": self.in_chs,
+                "out_chs": self.out_chs,
+                "exp_kernel_size": self.exp_kernel_size,
+                "stride": self.stride,
+                "dilation": self.dilation,
+                "group_size": self.group_size,
+                "pad_type": self.pad_type,
+                "force_in_chs": self.force_in_chs,
+                "noskip": self.noskip,
+                "exp_ratio": self.exp_ratio,
+                "pw_kernel_size": self.pw_kernel_size,
+                "act_layer": self.act_layer,
+                "norm_layer": self.norm_layer,
+                "se_layer": keras.saving.serialize_keras_object(self.se_layer),
+                "drop_path_rate": self.drop_path_rate,
+                "data_format": self.data_format,
+                "channel_axis": self.channel_axis,
+            }
+        )
+        return config
+
+    @classmethod
+    def from_config(cls, config):
+        config["se_layer"] = keras.saving.deserialize_keras_object(
+            config.pop("se_layer")
+        )
+        return cls(**config)
+
+
+class CondConvResidual(keras.layers.Layer):
+    """Conditionally Parameterized Convolutional Residual block.
+
+    This block uses a routing function to dynamically select and combine
+    different convolutional experts based on the input.
+
+    Args:
+        in_chs: int. The number of input channels.
+        out_chs: int. The number of output channels.
```

Comment:
`filters`, in general if you need the information about the input channel, you can always use something like `keras.ops.shape(x)[channel_axis]`